server/with-bun/utils/encryption-utils.ts

```diff
@@ -1,76 +1,95 @@
     throw new Error("ENCRYPTION_KEY must be set and be 32 bytes long");
   }
-
   return encryptionKey;
 }
 
-/**
- * Encrypts a given text using AES with CBC mode and PKCS7 padding.
- *
- * @param {string} text - The text to encrypt.
- * @returns {string} - The encrypted text in the format "IV:Ciphertext".
- */
-export function encrypt(text: string): string {
+async function importSecretKey(keyString: string): Promise<CryptoKey> {
+  try {
+    const keyBuffer = Buffer.from(keyString, "utf-8");
+    return await crypto.subtle.importKey("raw", keyBuffer, { name: ALGORITHM }, false, ["encrypt", "decrypt"]);
+  } catch (error) {
+    throw new Error(`Failed to import secret key: ${error instanceof Error ? error.message : String(error)}`);
+  }
+}
+
+export async function encrypt(text: string): Promise<string> {
   if (!text) {
     throw new Error("Text to encrypt must be provided");
   }
 
-  const iv = CryptoJS.lib.WordArray.random(16);
-  const key = CryptoJS.enc.Utf8.parse(getEncryptionKey());
+  try {
+    const keyString = getEncryptionKey();
+    const cryptoKey = await importSecretKey(keyString);
+    const iv = crypto.getRandomValues(new Uint8Array(IV_LENGTH));
+    const encodedText = new TextEncoder().encode(text);
 
-  const encrypted = CryptoJS.AES.encrypt(text, key, {
-    iv: iv,
-    mode: CryptoJS.mode.CBC,
-    padding: CryptoJS.pad.Pkcs7,
-  });
+    const encryptedBuffer = await crypto.subtle.encrypt({ name: ALGORITHM, iv: iv }, cryptoKey, encodedText);
 
-  return `${CryptoJS.enc.Hex.stringify(iv)}:${encrypted.ciphertext.toString()}`;
+    const ivBase64 = Buffer.from(iv.buffer).toString("base64");
+    const encryptedBase64 = Buffer.from(encryptedBuffer).toString("base64");
+
+    return `${ivBase64}:${encryptedBase64}`;
+  } catch (error) {
+    throw new Error(`Encryption process failed: ${error instanceof Error ? error.message : String(error)}`);
+  }
 }
 
-/**
- * Decrypts the provided encrypted text.
- *
- * @param {string} encryptedText - The encrypted text in the format "IV:Ciphertext".
- * @returns {string} - The decrypted plaintext.
- * @throws {Error} - If decryption fails.
- */
-export function decrypt(encryptedText: string): string {
+export async function decrypt(encryptedText: string): Promise<string> {
   if (!encryptedText || !encryptedText.includes(":")) {
-    throw new Error("Encrypted text must be in the format 'IV:Ciphertext'");
+    throw new Error("Encrypted text must be in the format 'IV(base64):Ciphertext(base64)'");
   }
 
-  const [ivHex, encryptedDataHex] = encryptedText.split(":");
-  const iv = CryptoJS.enc.Hex.parse(ivHex);
-  const encryptedData = CryptoJS.enc.Hex.parse(encryptedDataHex);
-  const key = CryptoJS.enc.Utf8.parse(getEncryptionKey());
+  const parts = encryptedText.split(":");
 
-  const cipherParams = CryptoJS.lib.CipherParams.create({
-    ciphertext: encryptedData,
-  });
+  if (parts.length !== 2 || !parts[0] || !parts[1]) {
```

Comment:
this block seems quite manual to have to manage in implementation code. why do we need it here?